nuitka/build/static_src/HelpersTypes.c

```diff
@@ -280,36 +288,73 @@ static PyTypeObject *_getTypeGenericAliasType(void) {
 
     if (type_generic_alias_type == NULL) {
 
-        PyObject *typing_module = PyImport_ImportModule("_typing");
-        CHECK_OBJECT(typing_module);
+        PyObject *types_module = PyImport_ImportModule("types");
+        CHECK_OBJECT(types_module);
 
-        type_generic_alias_type = (PyTypeObject *)PyObject_GetAttrString(typing_module, "_GenericAlias");
+        type_generic_alias_type = (PyTypeObject *)PyObject_GetAttrString(types_module, "GenericAlias");
         CHECK_OBJECT(type_generic_alias_type);
     }
 
     return type_generic_alias_type;
 }
 
+static int _Nuitka_contains_typevartuple(PyTupleObject *params) {
+    Py_ssize_t n = PyTuple_GET_SIZE(params);
+    PyTypeObject *tp = PyInterpreterState_Get()->cached_objects.typevartuple_type;
+    for (Py_ssize_t i = 0; i < n; i++) {
+        PyObject *param = PyTuple_GET_ITEM(params, i);
+        if (Py_IS_TYPE(param, tp)) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
 static PyObject *_Nuitka_unpack_typevartuples(PyObject *params) {
     assert(PyTuple_Check(params));
-
-    // TODO: Not implemented yet.
-
-    return Py_NewRef(params);
+    // TypeVarTuple must be unpacked when passed to Generic, so we do that here.
+    if (_Nuitka_contains_typevartuple((PyTupleObject *)params)) {
+        Py_ssize_t n = PyTuple_GET_SIZE(params);
+        PyObject *new_params = PyTuple_New(n);
```

Comment:
We have a function that is not API and creates a tuple with a size, using that instead. and also passing tstate in, as we have it on the call side, and getting it can only be expensive.